main.pl

```diff
@@ -211,10 +206,24 @@
 module_description("No description") -->
     ... .
 
+predicate_string(Predicate, Ops, PredicateString) :-
+    Predicate = PN/PA,
+    member(op(_, _, PN), Ops),
+    phrase(format_("(~a)/~d", [PN, PA]), PredicateString).
+
+predicate_string(Predicate, Ops, PredicateString) :-
+    Predicate = PN/_PA,
+    \+ member(op(_, _, PN), Ops),
```

Comment:
I think it's possible to avoid the case distinction by using the `~q` format specifier, and using `PN/PA` directly as the argument. For example, we get:

<pre>
?- Op = (+), Arity = 2, format("~q", [Op/Arity]).
<b>(+)/2</b>   Op = (+), Arity = 2.
</pre>

Reply:
But that wouldn't deal with the case that the predicate is an operator, and therefore needs to have parenthesis around them. It seems that `~q` only puts the parenthesis automatically with predicates that are defined as and operation in the current execution environment:

```
?- format("~q~n", [(===)/2]).
=== /2
   true. 
?- op(900, xfx, ===).
   true. 
?- format("~q~n", [(===)/2]).
(===)/2 
   true.
```

But here we aren't dealing with operators in the execution environment, but with operators in the libraries that were parsed. Just because a predicate is an operator when DocLog calls `format_//2` doesn't mean that it's an operator in the library it is documenting, and vice-versa.

It is probably possible to deal with this more elegantly and efficiently, but it probably will need to change the representation of the predicates in the entire program.

Reply:
Yes, either that, or load the library that's being documented.

Reply:
> load the library that's being documented.

I think this would have problems if two different libraries defined the same predicate. Especially if that predicate is an operator in one library but not an operator in another.

```
%% test_lib_1.pl
%:- module(test_lib_1, [
%    example/1
%]).
%
%example(_).

%% test_lib_2.pl
%:- module(test_lib_2, [
%    op(700, fx, example),
%    example/1
%]).
%
%example(_).

?- use_module(test_lib_1).
   true.
?- format("~q~n", [(example)/1]).
example/1
   true.
?- use_module(test_lib_2).
   true.
?- format("~q~n", [(example)/1]).
(example)/1
   true.
```

Maybe that could be avoided if there is a way to "undefine" operators. But even if there is, I think that modifying the database is not a very good way to do this. Reifying what predicates in a module are operators, which is what DocLog does now, seems less hacky and more declarative.